src/modules/fancyzones/FancyZonesLib/MonitorUtils.cpp

```diff
@@ -369,10 +355,20 @@ namespace MonitorUtils
 
         auto displays = Display::GetDisplays();
         auto monitors = WMI::GetHardwareMonitorIds();
+
+        // retry 
+        int retryCounter = 0;
+        while (!displays.has_value() && retryCounter < 100)
+        {
+            Logger::info("Retry display identification");
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+            displays = Display::GetDisplays();
+            retryCounter++;
+        }
         
         for (const auto& monitor : monitors)
         {
-            for (auto& display : displays)
+            for (auto& display : displays.value())
```

Comment:
displays can still be nullopt here, right? After 100 retries

Reply:
You're right!